runtime/decimal.c

```diff
@@ -107,6 +119,18 @@ TaggedPtrPanicCode finish(decQuad *dq, decContext *cx) {
     return result;
 }
 
+TaggedPtr _bal_decimal_neg(TaggedPtr tp) {
+    const decQuad *dq = taggedToDecQuad(tp);
+    decQuad d;
+    if (decQuadClass(dq) == DEC_CLASS_POS_ZERO) {
+        decQuadZero(&d);
```

Comment:
Better to return `tp` in this case.

Reply:
@jclark, If the input of `_bal_decimal_neg` is `0.0`, we have to return `0`. Please check the 2nd point of `Simplified number set` in doc http://speleotrove.com/decimal/dax3274.html
If we return `tp`, in this case we will get `0.0`.

Reply:
Please ignore my comment. `0.0` is not a valid case at this point, it should be handled before passing to this function.

Changed to return `tp`.